xdem/volume.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+
+def hypsometric_binning(ddem: np.ndarray, dem: np.ndarray):
+
+    assert ddem.shape == dem.shape
+    nan_mask = np.isnan(ddem) | np.isnan(dem)
+    ddem = ddem[~nan_mask]
+    dem = dem[~nan_mask]
+
+    old_dem = dem - ddem
+    elevation_range = np.percentile(old_dem, 99) - np.percentile(old_dem, 1)
```

Comment:
we need to find a way to be robust with the elevation range without removing any glacier area

Reply:
What do you mean? The elevation range is only used for calculating 10% steps. The actual bins are generated from `old_dem.min()` to `old_dem.max() + `bin_size`